src/main/java/de/tum/in/www1/artemis/service/export/ProgrammingExerciseExportService.java

```diff
@@ -142,6 +147,15 @@ private Path exportProgrammingExerciseMaterialWithStudentReposOptional(Programmi
                     pathsToBeZipped.add(path);
                 }
             });
+
+            // Export the build plan of a programming exercise, if one exists. Only relevant for Gitlab/Jenkins or Gitlab/GitlabCI setups.
+            var buildPlan = buildPlanRepository.findByProgrammingExercises_IdWithProgrammingExercises(exercise.getId());
```

Comment:
```suggestion
            var buildPlan = buildPlanRepository.findByProgrammingExercises_Id(exercise.getId());
```
An additional query in the repository could avoid fetching data we do not need anyway.